include/deal.II/matrix_free/shape_info.templates.h

```diff
@@ -316,54 +316,91 @@ namespace internal
 
           try
             {
-              const auto reference_cell_type = ReferenceCell::get_simplex(dim);
+              const auto reference_cell_type = fe.reference_cell_type();
 
-              const auto quad_face  = get_face_quadrature(quad);
-              this->n_q_points_face = quad_face.size();
+              const auto temp      = get_face_quadrature_collection(quad);
+              const auto quad_face = temp.second;
 
-              const unsigned int n_face_orientations = dim == 2 ? 2 : 6;
-              const unsigned int n_faces =
-                ReferenceCell::internal::Info::get_cell(reference_cell_type)
-                  .n_faces();
+              AssertThrow(reference_cell_type == temp.first,
+                          ExcNotImplemented());
+
+              this->n_q_points_face = quad_face[0].size();
+
+              // TODO: lift limitation
+              for (unsigned int i = 1; i < quad_face.size(); ++i)
+                AssertThrow(this->n_q_points_face == quad_face[i].size(),
+                            ExcNotImplemented());
```

Comment:
This is the place where some (minor) changes have to be done for wedges and pyramids. I'll address this once #11361 and this PR are meged.